### PR TITLE
MM-8840: Add GlobalRelay Configuration Settings

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
Added system console integration for new GlobalRelay config values. These settings will be used as a part of [MM-8838](https://mattermost.atlassian.net/browse/MM-8838).

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-8840

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has [server changes](https://github.com/mattermost/mattermost-server/pull/8304)
- [x] Has [enterprise changes](https://github.com/mattermost/enterprise/pull/266)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

**Comment by jespino** [comment]
Now is ready to be reviewed :)

**Comment by mkraft** [comment]
Done.

File: i18n/en.json [hunk 1]
```diff
@@ -203,7 +203,6 @@
...
   "admin.complianceExport.createJob.title": "Run Compliance Export Job Now",
   "admin.complianceExport.description": "This feature supports compliance exports to the Actiance XML and GlobalRelay EML formats, and is currently in beta. Support for the Mattermost CSV format is scheduled for a future release, and will replace the existing <a href=\"/admin_console/general/compliance\">Compliance</a> feature.",
-  "admin.complianceExport.description_without_globalrelay": "This feature supports compliance exports to the Actiance XML format, and is currently in beta. Support for GlobalRelay EML and the Mattermost CSV formats is scheduled for a future release, and will replace the existing <a href=\"/admin_console/general/compliance\">Compliance</a> feature.",
   "admin.complianceExport.exportFormat.actiance": "Actiance XML",
   "admin.complianceExport.exportFormat.actiance.description": "Format of the compliance export. Corresponds to the system that you want to import the data into. Compliance Export files will be written to the \"exports\" subdirectory of the configured <a href=\"/admin_console/files/storage\">Local Storage Directory</a>.",
```
Comment (jasonblais) [inline i18n/en.json]: It's a bit confusing that the setting help text changes based on the setting value (different help text if Actiance or Global Relay selected).

Propose instead having a static help text with

```
Format of the compliance export. Corresponds to the system that you want to import the data into. 

For Actiance XML, compliance export files are written to the "exports" subdirectory of the configured [Local Storage Directory](link). For Global Relay EML, they are emailed to the configured email address.
```

File: i18n/en.json [hunk 2]
```diff
@@ -212,9 +211,20 @@
   "admin.complianceExport.exportJobStartTime.description": "Set the start time of the daily scheduled compliance export job. Choose a time when fewer people are using your system. Must be a 24-hour time stamp in the form HH:MM.",
   "admin.complianceExport.exportJobStartTime.example": "E.g.: \"02:00\"",
   "admin.complianceExport.exportJobStartTime.title": "Compliance Export time:",
+  "admin.complianceExport.globalRelaySettings.title": "GlobalRelay Settings:",
```
Comment (jasonblais) [inline i18n/en.json]: Propose we remove this. It's inconsistent with our other pages

image

File: i18n/en.json [hunk 3]
```diff
@@ -212,9 +211,20 @@
...
+  "admin.complianceExport.globalRelayCustomerType.title": "Customer Type:",
```
Comment (jasonblais) [inline i18n/en.json]: Proposed UI texts for the settings

```
Title: Global Relay Customer Account
Options: A9/Type 9; A10/Type 10
Help text: Type of Global Relay customer account your organization has.

Title: Global Relay SMTP Username
Help text: The username for authenticating to the Global Relay SMTP server.

Title: Global Relay SMTP Password
Help text: The password associated with the SMTP username.

Title: Global Relay Email Address
Help text: The email address your Global Relay server monitors for incoming compliance exports.

**Comment by jasonblais** [comment]
Sorry it took so long to review this. Let me know if there are any questions. Maybe @mkraft you can help with the updates?

**Comment by mkraft** [comment]
@jasonblais I'll ping you when the changes are ready.

**Comment by mkraft** [comment]
@jasonblais Changes pushed. Here's what it looks like now. Let me know if there are any other changes.

screen shot 2018-03-12 at 9 46 58 am
screen shot 2018-03-12 at 9 46 47 am 1

**Comment by jasonblais** [comment]
Thanks @mkraft! Looks good, can we just get rid of the `Global Relay Settings` section header as it's inconsistent with all our other System Console pages. I think you might have missed that note above.

image

**Comment by jespino** [comment]
Removed the header

**Comment by jasonblais** [comment]
Thanks! Also removed it from en.json file. Looks good to me 👍

**Comment by jasonblais** [comment]
@jespino @mkraft Anything blocking the merge of this PR?

**Comment by jespino** [comment]
Not really, I rebased it this morning.

**Pull request merged**